app/grandchallenge/algorithms/urls.py

```diff
@@ -133,7 +134,16 @@
         name="model-update",
     ),
     path("<slug>/jobs/", JobsList.as_view(), name="job-list"),
-    path("<slug>/jobs/create/", JobCreate.as_view(), name="job-create"),
+    path(
+        "<slug:slug>/jobs/interface-select/",
+        JobInterfaceSelect.as_view(),
+        name="job-interface-select",
+    ),
+    path(
+        "<slug:slug>/<uuid:interface>/jobs/create/",
```

Comment:
I think this would help clarity as you're passing a pk around and not an instance:

```suggestion
        "<slug:slug>/<uuid:interface_pk>/jobs/create/",
```

Reply:
Good point. And I actually think that `slug` should be changed to `algorithm_slug` for clarity. Otherwise it's unclear what object's slug this refers to.

Reply:
On second thought I will leave `slug` as is, because changing it to `algorithm_slug` here would require doing it for all other urls in the file - not worth it, and it's also actually pretty clear that the slug refers to the algorithm. Don't know why I was confused before.